pkg/linters/components/compimpl/linter.go

```diff
@@ -0,0 +1,164 @@
+	return nil, nil
+}
+
+func displayDiagnostic(pass *analysis.Pass, compimpls map[string]*implFolder) {
+	for folderName, implFolder := range compimpls {
+		file := implFolder.file
+		if !implFolder.hasProvides {
+			pass.Report(analysis.Diagnostic{
+				Pos:     file.Pos(),
+				Message: missingProvides(folderName),
+			})
+		}
+		if !implFolder.HasRequires {
+			pass.Report(analysis.Diagnostic{
+				Pos:     file.Pos(),
+				Message: missingRequires(folderName),
+			})
+		}
+		if !implFolder.HasConstructor {
+			pass.Report(analysis.Diagnostic{
+				Pos:     file.Pos(),
+				Message: missingConstructor(folderName),
+			})
+		}
+	}
+}
+
+func tryGetCompImpl(
+	pass *analysis.Pass,
+	file *ast.File,
+	isCompRegexp *regexp.Regexp,
+	compimpls map[string]*implFolder) *implFolder {
+	path := pass.Fset.Position(file.Pos()).Filename
+	capture := isCompRegexp.FindStringSubmatch(path)
+	if len(capture) != 2 {
+		return nil
+	}
+	compimplPath := filepath.Clean(capture[1])
+	if _, ok := compimpls[compimplPath]; !ok {
+		compimpls[compimplPath] = &implFolder{file: file}
+	}
+
+	return compimpls[compimplPath]
+}
+
+func missingProvides(folderName string) string {
+	return missingEntity(folderName, "type 'Provides'")
+}
+
+func missingEntity(folderName string, missingEntityMsg string) string {
+	return fmt.Sprintf("%v does not follow the component naming convention. It doesn't contain the %v", folderName, missingEntityMsg)
+}
+
+func missingRequires(folderName string) string {
+	return missingEntity(folderName, "type 'Requires'")
+}
+
+func missingConstructor(folderName string) string {
+	return missingEntity(folderName, "constructor of the component doesn't take as parameter 'Provides' and doesn't return 'Requires'")
+}
+
+func getExportedStructTypeNames(f *ast.File) map[string]bool {
+	structTypes := make(map[string]bool)
+	ast.Inspect(f, func(node ast.Node) bool {
+		switch t := node.(type) {
+		case *ast.TypeSpec:
+			if t.Name.IsExported() {
+				_, ok := t.Type.(*ast.StructType)
+				if ok {
+					structTypes[t.Name.Name] = true
+				}
+			}
+		}
+		return true
+	})
+	return structTypes
+}
+
+func hasComponentConstructor(f *ast.File) bool {
+	foundConstructor := false
+	ast.Inspect(f, func(node ast.Node) bool {
+		funcDecl, ok := node.(*ast.FuncDecl)
+		if !ok {
+			return true
+		}
+
+		if !strings.HasPrefix(funcDecl.Name.Name, "New") {
+			return true
+		}
+
+		params := funcDecl.Type.Params.List
+		results := funcDecl.Type.Results.List
+		if len(params) != 1 || len(results) < 1 || len(results) > 2 {
+			return true
+
+		}
+		if containFieldsType(params, "Requires") && containFieldsType(results, "Provides") {
+			foundConstructor = true
+			return false
+		}
+
+		return false
+	})
+	return foundConstructor
+}
+
+func containFieldsType(fields []*ast.Field, typeName string) bool {
+	for _, field := range fields {
+		ident, ok := field.Type.(*ast.Ident)
+		if !ok {
+			continue
+		}
+		if ident.Name == typeName {
+			return true
+		}
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏You can remove one if block here
```suggestion
		if ok && ident.Name == typeName {
			return true
		}
```